src/einsteinpy/rays/shadow.py

```diff
@@ -0,0 +1,96 @@
+import warnings
+
+import astropy.units as u
+import numpy as np
+from scipy.integrate import quadrature
+from scipy.interpolate import interp1d
+from scipy.optimize import newton
+
+
+class Shadow:
+    """
+    Class for calculating shadow of black holes as seen by distant observer.
+    """
+
+    @u.quantity_input(
+        mass=u.kg, fov=u.km,
+    )
+    def __init__(self, mass, n_rays, fov, limit=0.001):
+        self.mass = mass
```

Comment:
```suggestion
        self.mass = mass.to(u.kg)
```

`u.quanity_input` does not take care of automatic conversion of units. It just checks the compatibility.

Reply:
Yeah